### PR TITLE
Republic

### PR DESCRIPTION
**Pull request by contextlimit**
Status: Do not merge

### CODE REVIEW

File: smart-contracts/ethereum/contracts/Republic.sol [hunk 1]
```diff
@@ -22,27 +22,54 @@ contract Republic {
...
+    return true;
   }
 
   function getDelegateAddresses() public view returns(address[11] res) {
     return delegateAddresses;
   }
 
-  function createVote() onlyDelegate public view returns(bool res) { // TODO: string name, uint version, string files, string checksum
-    return true;
+  function startElection() onlyDelegate public view returns(RepublicPrimaryElection res) { // TODO: string name, uint version, string files, string checksum
```
Comment (contextlimit) [inline smart-contracts/ethereum/contracts/Republic.sol]: Remove TODO

File: smart-contracts/ethereum/contracts/Republic.sol [hunk 2]
```diff
@@ -22,27 +22,54 @@ contract Republic {
...
+    currentElectionIndex = electionAddresses.length - 1;
+    elections[election] = true;
+
+    election.start();
+    
+    return election;
   }
 
-  function endVote() onlyDelegate public view returns(bool res) { // TODO: uint id, bool vote
+  function endElection() onlyDelegate public view returns(bool res) { // TODO: uint id, bool vote
```
Comment (contextlimit) [inline smart-contracts/ethereum/contracts/Republic.sol]: Remove TODO

File: smart-contracts/ethereum/contracts/Republic.sol [hunk 3]
```diff
@@ -22,27 +22,54 @@ contract Republic {
...
+
+    uint256 electionResult = currentElection.tallyVotes();
+
+    if (electionResult == 1) {
+        // Successful election
+
+    } else {
+        // Log unsuccessful election reason
+
+    }
```
Comment (contextlimit) [inline smart-contracts/ethereum/contracts/Republic.sol:105]: Dont forget this

File: smart-contracts/ethereum/contracts/RepublicIndustryElection.sol [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+  string industry;
+  bool started;
+  
+  function RepublicIndustryElection(string _industry) public {
+    started = false;
+    delegates[msg.sender] = true;
+    industry = _industry;
+  }
+
+  // TODO: add onlyDelegate modifier
```
Comment (contextlimit) [inline smart-contracts/ethereum/contracts/RepublicIndustryElection.sol]: Dont forget this

File: smart-contracts/ethereum/contracts/RepublicPrimaryElection.sol [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+      industryElections[i].start();
+    }
+
+    return true;
+  }
+
+  function tallyVotes() public payable returns(uint256 res) {
+
+    return 1;
+  }
```
Comment (contextlimit) [inline smart-contracts/ethereum/contracts/RepublicPrimaryElection.sol]: Dont forget this

**Comment by WalidOfNow** [comment]
Merged

**Pull request merged**